app_dart/lib/src/request_handlers/check_for_waiting_pull_requests.dart

```diff
@@ -0,0 +1,280 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert' show jsonEncode;
+
+import 'package:appengine/appengine.dart';
+import 'package:cocoon_service/src/request_handling/exceptions.dart';
+import 'package:graphql/client.dart';
+import 'package:meta/meta.dart';
+
+import '../datastore/cocoon_config.dart';
+import '../foundation/providers.dart';
+import '../foundation/typedefs.dart';
+import '../request_handling/api_request_handler.dart';
+import '../request_handling/authentication.dart';
+import '../request_handling/body.dart';
+
+import 'check_for_waiting_pull_requests_queries.dart';
+
+@immutable
+class CheckForWaitingPullRequests extends ApiRequestHandler<Body> {
+  const CheckForWaitingPullRequests(
+    Config config,
+    AuthenticationProvider authenticationProvider, {
+    @visibleForTesting LoggingProvider loggingProvider,
+  })  : loggingProvider = loggingProvider ?? Providers.serviceScopeLogger,
+        super(config: config, authenticationProvider: authenticationProvider);
+
+  final LoggingProvider loggingProvider;
+
+  @override
+  Future<Body> get() async {
+    final Logging log = loggingProvider();
+    final GraphQLClient client = await config.createGitHubGraphQLClient();
+
+    await _checkPRs('flutter', 'flutter', log, client);
+    await _checkPRs('flutter', 'engine', log, client);
+
+    return Body.empty;
+  }
+
+  Future<void> _checkPRs(
+    String owner,
+    String name,
+    Logging log,
+    GraphQLClient client,
+  ) async {
+    bool hasMerged = false;
+    final Map<String, dynamic> data = await _queryGraphQL(
+      owner,
+      name,
+      log,
+      client,
+    );
+    for (_AutoMergeQueryResult queryResult in _parseQueryData(data)) {
+      if (!hasMerged && queryResult.shouldMerge) {
+        hasMerged = await _mergePullRequest(
```

Comment:
hasMerged is not used.

Reply:
Ah thanks - at one point I was returning whether the merge succeeded.

Reply:
Actually it is used - we stop looking once we merged.

Reply:
Oh, right.